src/Adapter/MSTest.CoreAdapter/Execution/TestClassInfo.cs

```diff
@@ -298,19 +340,26 @@ public void RunClassInitialize(TestContext testContext)
         /// <returns>
         /// Any exception that can be thrown as part of a class cleanup as warning messages.
         /// </returns>
-        [SuppressMessageAttribute("Microsoft.Design", "CA1031:DoNotCatchGeneralExceptionTypes", Justification = "Requirement is to handle all kinds of user exceptions and message appropriately.")]
+        [SuppressMessage("Microsoft.Design", "CA1031:DoNotCatchGeneralExceptionTypes", Justification = "Requirement is to handle all kinds of user exceptions and message appropriately.")]
         public string RunClassCleanup()
         {
-            if (this.ClassCleanupMethod == null)
+            if (this.ClassCleanupMethod is null && !this.BaseClassInitAndCleanupMethods.Any(p => p.Item2 != null))
```

Comment:
not that this would affect the code here but [this](https://stackoverflow.com/questions/40676426/what-is-the-difference-between-x-is-null-and-x-null) is an interesting discussion on the difference between `== null` and `is null`

Reply:
Indeed, pretty interesting discussion. When I first saw the new (not that new already) `is` expression I read some posts about what differentiates it from the `==` on null comparison and was kind of similar to this discussion except for the part that they didn't show how resulted IL code is. Thanks for sharing that out :)

And yeah, as you first pointed out this wouldn't affect because no equality overloaded is in the code checking if the type we're is null.